src/store/reducers/index.js

```diff
@@ -120,6 +122,18 @@ export const reducer = (state = initialState, action) => {
         individualPostCommentsAreFetching: false,
       };
 
+    case 'SET_DRAWER_VISIBILITY':
+      return {
+        ...state,
+        isDrawerVisible: action.payload,
+      };
+
+    case 'SET_NEW_ROOM_MODAL_VISIBILITY':
+      return {
+        ...state,
+        isNewRoomModalVisible: action.payload,
+      };
+
```

Comment:
Same with my store comment, this is good enough to pass, but with `combineReducers` implemented, this should be placed in a `modalReducer.js` file. Nonetheless I like the idea of using redux here, curious what reason you had to not manage this at the component level.

Reply:
At first I was keeping it at the component level but as I was refactoring, I wanted to separate the modal from the component that activates it. I could've passed things through problem but I find that it gets messy when you have to pass callback functions and since we already had store set up I figured I would use it. I would really like to take some time to refactor this, the only problem is that using combine reducers would mean having to go through every `mapPropsToState` function and changing the path for the states needed. I think that's the main reason we haven't refactored the store yet

Reply:
100% feel you on store refactor. Seems like it should be a roadmap goal for a future cohort and maybe we just provide some surface level direction in our notes. Thanks for taking the time to explain! Definitely makes sense.